pkg/receive/handler.go

```diff
@@ -403,6 +405,13 @@ func (h *Handler) handleRequest(ctx context.Context, rep uint64, tenant string,
 	return h.forward(ctx, tenant, r, wreq)
 }
 
+func (h *Handler) isTenantValid(tenant string, err error, w http.ResponseWriter) {
+	if tenant != path.Base(tenant) {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
```

Comment:
Maybe we can return an error here, and then log + respond 500 in receive HTTP instead? 🙂

Reply:
+1, will make the changes, it'll make testing easier. Will push a draft so we can iterate